### ISSUE TITLE
Shouldn't this be a compare and swap?

### ISSUE THREAD

**Issue by kiranandcode** [koonwen/flat-combiner#1]
https://github.com/koonwen/flat-combiner/blob/4a9a2df618cfcc08880f69a09694c052166a67ba/src/fc_generic/fc_generic_domains.ml#L64

**Comment by koonwen** [comment]
Yup it should be, thanks for pointing it out! I think I wrote it this way just to keep it simple for the first attempt. I'll change it soon but as of now I don't think it adds very much overhead without CAS.
